### ISSUE TITLE
Gradle: Deprecation of maven plugin in Gradle 6.0 and removal with 7.0

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#1547]
> The maven plugin has been deprecated. This is scheduled to be removed in Gradle 7.0. Please use the maven-publish plugin instead.

we make have use of the maven plugin in our gradle builds and especially in the publishing plugin
=> we should start having a look how to replace that.

for the normal plugins you can find a indication on the https://github.com/eclipse/xtext-core/tree/cd_javaPlatformExperiments branch.

what we can do about the publishing? no idea

**Comment by cdietrich** [comment]
@kthoms is there a possibility to verify if sonatype oss likes the poms/jar/javadocs or does this work on master/release branches only?

**Comment by cdietrich** [comment]
one difference is that the `maven-publish` plugin seems to write test scope dependencies to the pom.
this should not harm, will check if this can be avoided anyway

**Comment by cdietrich** [comment]
it looks like the publishing plugin is already using the maven-publish plugin so we may safe there. 
but there are other problems like

```
> Could not create task ':downloadXtextP2Repository'.
   > Could not create task of type 'Download'.
      > org.gradle.api.tasks.TaskInputs.property(Ljava/lang/String;Ljava/lang/Object;)Lorg/gradle/api/tasks/TaskInputs;

```

**Comment by cdietrich** [comment]
Fixed in 2.20
